### ISSUE TITLE
Add new scanner action that delays indefinitely and restrict the authors

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#16504]
We already have a scanner rule action that delays indefinitely, we want one that also automatically restrict the associated users by email and IP.

**Comment by AlexandraMoga** [comment]
I've verified this issue on stage by setting a scanner rule action to 'Delay auto-approval indefinitely and add restrictions' and then submitting an add-on that would match that rule. Both the user email and the IP have been added to the Email user restrictions, respectively IP network user restrictions. The user is blocked from submitting further add-ons from the same email address. Trying to submit an add-on from the same IP but with a different user is also not allowed.

If there are multiple scanner rules active, which also have a Delay approval action, the 'Delay auto-approval indefinitely and add restrictions' takes precedence when an add-on matches all those rules.

@diox I've noticed the following behavior for IP restrictions and I don't remember if it is expected or not:
- user logs in from a restricted IP
- user tries to submit an add-on => submission is blocked
- user changes the IP address (using a VPN) but doesn't log out
- user tries to submit an add-on again => submission is blocked

I suppose this happens because the user's Last Known IP address recorded at our end is still the restricted one, even if the user changed it in the meantime. The IP restriction is removed only after the user logs out and logs in again.
